samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -37,39 +37,147 @@ static K_SEM_DEFINE(sem_stream_qos, 0, 1);
 static K_SEM_DEFINE(sem_stream_enabled, 0, 1);
 static K_SEM_DEFINE(sem_stream_started, 0, 1);
 
-void print_hex(const uint8_t *ptr, size_t len)
+
+#if defined(CONFIG_LIBLC3CODEC)
+
+#include "lc3.h"
+#include "math.h"
+
+/* Current sample do not use codec configuration parameters, hence below shall match the selected
+ * codec configuration.
+ * One sample data buffer en generated and repeated.
+ */
+#define AUDIO_SAMPLE_RATE_HZ    16000
+#define AUDIO_FREQUENCY_HZ      400
+#define AUDIO_LENGTH_US         10000 /* amount of sample data - shall match LC3 frame length */
+#define AUDIO_LENGTH_100US      (AUDIO_LENGTH_US / 100)
+#define NUM_SAMPLES             ((AUDIO_LENGTH_US * AUDIO_SAMPLE_RATE_HZ) / USEC_PER_SEC)
+#define AUDIO_VOLUME            (INT16_MAX - 3000) /* codec does clipping above INT16_MAX - 3000 */
+
+static int16_t audio_buf[NUM_SAMPLES];
+static lc3_encoder_t lc3_encoder;
+static lc3_encoder_mem_48k_t lc3_encoder_mem;
+
+
+/* Some of the prints made from this application take too long causing the applicaiton to fail.
+ * Enable only when the debug terminal can handle it - e.g. baudrate > 115200
+ */
+#define VERBOSE false
```

Comment:
You can add this a sample Kconfig option. I think that's more useful than a hardcoded macro, and makes it easier to compile with this value set to different values (as it doesn't require a source code change).

Reply:
I don't agree :-) I find it much easier to locate local (single file) macroconfigs when they are in the top of a file rather than I need to locate the KConfig to find the name of the macro, open menuconfig to apply the change and rebuild the full app.
I'll just remove the log statement and this macro all together.